ASSERT while clearing ASTC textures for copy

kkinnunen-apple opened the pull request:
#### 71bd202cfeb44ab86897267afd37bd7bda0cb17a
<pre>
ASSERT while clearing ASTC textures for copy
<a href="https://bugs.webkit.org/show_bug.cgi?id=273099">https://bugs.webkit.org/show_bug.cgi?id=273099</a>
<a href="https://rdar.apple.com/126894196">rdar://126894196</a>

Reviewed by Mike Wyrzykowski.

Textures would be cleared with blit from empty texture. The blit size
would be calculated from logical texture size. This is incorrect for
cases where compressed texture mipmap levels != 1 would be used, since
their logical sizes might not be block aligned. The physical sizes
would need to be used.

Use correct function WTF::roundUpToMultipleOfNonPowerOfTwo().

* LayoutTests/fast/webgpu/copy-texture-to-texture-astc-mipmap-dimensions-crash-expected.txt: Added.
* LayoutTests/fast/webgpu/copy-texture-to-texture-astc-mipmap-dimensions-crash.html: Added.
* Source/WebGPU/WebGPU/CommandEncoder.mm:
(WebGPU::CommandEncoder::clearTextureIfNeeded):
* Source/WebGPU/WebGPU/Texture.h:
* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::Texture::bytesPerRow):
(WebGPU::Texture::physicalMiplevelSpecificTextureExtent):
(WebGPU::Texture::physicalTextureExtent):

Canonical link: <a href="https://commits.webkit.org/278006@main">https://commits.webkit.org/278006@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/536ed4b327d7e6e1d9b29fa4a3afb87305536654

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/48855 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/28068 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/51822 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/51542 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/44921 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/34018 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/25596 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/39943 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/49437 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/25721 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/42131 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/21045 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/23184 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/43305 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/6910 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/45115 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/43801 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/53453 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/23906 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/20175 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/47244 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/25169 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/42340 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/46193 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10821 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/25977 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/24889 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski reviewed (approved):
Thank you for fixing this!

kkinnunen-apple committed:
ASSERT while clearing ASTC textures for copy
https://bugs.webkit.org/show_bug.cgi?id=273099
rdar://126894196

Reviewed by Mike Wyrzykowski.

Textures would be cleared with blit from empty texture. The blit size
would be calculated from logical texture size. This is incorrect for
cases where compressed texture mipmap levels != 1 would be used, since
their logical sizes might not be block aligned. The physical sizes
would need to be used.

Use correct function WTF::roundUpToMultipleOfNonPowerOfTwo().

* LayoutTests/fast/webgpu/copy-texture-to-texture-astc-mipmap-dimensions-crash-expected.txt: Added.
* LayoutTests/fast/webgpu/copy-texture-to-texture-astc-mipmap-dimensions-crash.html: Added.
* Source/WebGPU/WebGPU/CommandEncoder.mm:
(WebGPU::CommandEncoder::clearTextureIfNeeded):
* Source/WebGPU/WebGPU/Texture.h:
* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::Texture::bytesPerRow):
(WebGPU::Texture::physicalMiplevelSpecificTextureExtent):
(WebGPU::Texture::physicalTextureExtent):

Canonical link: https://commits.webkit.org/278006@main
